### PR TITLE
Miscellaneous improvement

### PR DESCRIPTION
**Pull request by andylokandy**

### CODE REVIEW

**Comment by andylokandy** [comment]
Next stage of work would be search support and front-end implementation. To be honest, I'm not so confident on building a real front-end project. My last project is in two years ago, based on angular 1.0, which quite outdated. I think we need more contributors to help with it.

**Comment by dcao** [comment]
Yeah, I think once we get the backend done I'll try to get some help on the frontend-side of things (I know a friendly redditor wanted to help out in this part)

**Comment by andylokandy** [comment]
Is this PR good to merge?

File: src/index/mod.rs [hunk 1]
```diff
@@ -149,12 +154,58 @@ impl Handler<YankAndUpdate> for Index {
         self.repo
             .commit_and_push(
                 &format!(
-                    "Updating package `{}/{}#{}`",
-                    &msg.package.name.group, &msg.package.name.name, &msg.package.semver
+                    "Updating package `{}#{}`",
```
Comment (dcao) [inline src/index/mod.rs]: Right now we use `package_id|version` instead of the hash I think

**Pull request merged**